src/reducers/editThought.ts

```diff
@@ -32,6 +45,28 @@ const editThought = (
 
   const editedThought = getThoughtById(state, editedThoughtId)
 
+  const parentThoughtIdForEditedThought = parentOfThought(state, editedThoughtId)?.id
+
+  // only calculate decendant thought when current edited thought is a metaprogramming attribute
+
+  const thoughtIdForExistingMetaProgrammingThought =
```

Comment:
You should also check that the cursor is on the edited thought. `editThought` is called within various reducer flows, so you can't assume the edited thought is the cursor.